my_file.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+if [ -n "$1" ]; then
+num=$1
+else 
+num=50
+fi
+d=$(date +'%d_%m_%y')
+mkdir "zero_$(echo $d)_files"
+cd "zero_$(echo $d)_files"
```

Comment:
you use $(echo $d) to substitute the value of the variable $d. However, since the variable doesn't contain any special characters, you can directly use $d without command substitution.